channel/player.go

```diff
@@ -1059,6 +1081,27 @@ func (d player) displayBytes() []byte {
 	return pkt
 }
 
+// Logout flushes coalesced state and does a full checkpoint save.
+func (d *player) Logout(reason string) {
+	log.Printf("player(%d) logout: %s", d.id, reason)
```

Comment:
I think we should avoid writing to stdout as much as possible (preferebly just errors)